tests/test_s3tables/test_s3tables.py

```diff
@@ -0,0 +1,424 @@
+    client.create_table(
+        tableBucketARN=arn, namespace="bar", name="baz", format="ICEBERG"
+    )
+
+    resp = client.list_tables(tableBucketARN=arn, namespace="bar")
+    assert resp["tables"]
+    assert resp["tables"][0]["name"] == "baz"
+
+
+@mock_aws
+def test_list_tables_pagination():
+    client = boto3.client("s3tables", region_name="us-east-2")
+    arn = client.create_table_bucket(name="foo")["arn"]
+    client.create_namespace(tableBucketARN=arn, namespace=["bar"])
+    client.create_table(
+        tableBucketARN=arn, namespace="bar", name="baz", format="ICEBERG"
+    )
+    client.create_table(
+        tableBucketARN=arn, namespace="bar", name="baz2", format="ICEBERG"
+    )
+
+    resp = client.list_tables(tableBucketARN=arn, maxTables=1)
+    assert len(resp["tables"]) == 1
+    assert "continuationToken" in resp
+
+    resp = client.list_tables(
+        tableBucketARN=arn, maxTables=1, continuationToken=resp["continuationToken"]
+    )
+
+    assert len(resp["tables"]) == 1
+    assert "continuationToken" not in resp
+    assert resp["tables"][0]["name"] == "baz2"
+
+
+@mock_aws
+def test_delete_table():
+    client = boto3.client("s3tables", region_name="us-east-2")
+    arn = client.create_table_bucket(name="foo")["arn"]
+    client.create_namespace(tableBucketARN=arn, namespace=["bar"])
+    client.create_table(
+        tableBucketARN=arn, namespace="bar", name="baz", format="ICEBERG"
+    )
+
+    resp = client.list_tables(tableBucketARN=arn)
+    assert len(resp["tables"]) == 1
+
+    client.delete_table(tableBucketARN=arn, namespace="bar", name="baz")
+
+    resp = client.list_tables(tableBucketARN=arn)
+    assert len(resp["tables"]) == 0
+
+
+@mock_aws
+def test_delete_table_deletes_underlying_table_storage():
+    client = boto3.client("s3tables", region_name="us-east-2")
+    s3 = boto3.client("s3", region_name="us-east-2")
+    arn = client.create_table_bucket(name="foo")["arn"]
+    client.create_namespace(tableBucketARN=arn, namespace=["bar"])
+    client.create_table(
+        tableBucketARN=arn, namespace="bar", name="baz", format="ICEBERG"
+    )
+    warehouse = client.get_table(tableBucketARN=arn, namespace="bar", name="baz")[
+        "warehouseLocation"
+    ]
+
+    bucket_name = warehouse.replace("s3://", "")
+    s3.head_bucket(Bucket=bucket_name)
+
+    client.delete_table(tableBucketARN=arn, namespace="bar", name="baz")
+    with pytest.raises(s3.exceptions.ClientError):
+        s3.head_bucket(Bucket=bucket_name)
+
+
+@mock_aws
+def test_delete_table_fails_on_version_token_mismatch():
+    client = boto3.client("s3tables", region_name="us-east-2")
+    arn = client.create_table_bucket(name="foo")["arn"]
+    client.create_namespace(tableBucketARN=arn, namespace=["bar"])
+    resp = client.create_table(
+        tableBucketARN=arn, namespace="bar", name="baz", format="ICEBERG"
+    )
+    token = resp["versionToken"]
+
+    with pytest.raises(client.exceptions.ConflictException):
```

Comment:
Would be useful to also match on the error, to verify we're throwing the correct error here